cmd/kind/load/docker-image/docker-image.go

```diff
@@ -88,18 +98,37 @@ func runE(flags *flagpole, cmd *cobra.Command, args []string) error {
 
 	// pick only the user selected nodes and ensure they exist
 	// the default is all nodes unless flags.Nodes is set
-	selectedNodes := nodes
+	candidateNodes := nodes
 	if len(flags.Nodes) > 0 {
-		selectedNodes = []clusternodes.Node{}
+		candidateNodes = []clusternodes.Node{}
 		for _, name := range flags.Nodes {
 			node, ok := nodesByName[name]
 			if !ok {
-				return errors.Errorf("unknown node: %s", name)
+				return errors.Errorf("unknown node: %q", name)
 			}
+			candidateNodes = append(candidateNodes, node)
+		}
+	}
+
+	// pick only the nodes that don't have the image
+	selectedNodes := []clusternodes.Node{}
+	for _, node := range candidateNodes {
+		// Don't load image on external nodes like the load balancer
+		if strings.Contains(node.String(), "external") {
+			continue
+		}
+
+		cmdNode := node.Command("crictl", "-r", "/var/run/containerd/containerd.sock", "inspecti", imageName)
```

Comment:
is this ok?

Reply:
Put a TODO to move it out at least, cmd/ should only have application boilerplate, pkg/... should be providing this logic

Reply:
done